app/modules/auth/auth.factory.js

```diff
@@ -22,18 +22,22 @@ function authFactory($location, $rootScope, $state, $transitions, AnnotationsFac
             zooAPI.beforeEveryRequest = function() {
                 return OAuth.checkBearerToken()
                     .then(function (token) {
-                        // The Panoptes client doesn't return an error but just null when it can't refresh the token.
-                        // So we check for null, instead of using a catch block.
-
-                        if (_user.id && token === null) {
-                            // We are logged in but don't have a token any more.
-                            alert('Your session is expired. Press OK to save your work and start a new one.')
-                            // Save any unsaved work and redirect to Panoptes for a new token.
-                            // AnnotationsFactory.updateCache();
-                            // OAuth.signIn($location.absUrl());
-                            return Promise.reject(new Error('HELP!'));
-
-                        }
+                        // We catch any request to access the transcribe route.
+                        $transitions.onStart({ to: 'Transcribe'}, function(transition) {
+                            token = $window.sessionStorage.getItem('panoptesClientOAuth_tokenDetails');
```

Comment:
Never, ever access the client's storage directly like this. You should always go through `OAuth` in order to validate the session.

Reply:
I added that like because `token` is still defined even when storage is cleared. Would `var cachedToken = $window.sessionStorage.getItem('panoptesClientOAuth_tokenDetails');` be better?

Reply:
Absolutely not. Don't load the client's token directly because a) you don't know what name the client code uses internally the store the token, and this may well change b) the client has a public method defined to get the token for you: `checkBearerToken`.

Reply:
Clearing storage logs your session out, so the client logs you back in if you still have a valid Panoptes session. It doesn't simulate your session having expired.

Reply:
Thanks for clarify. What's the alternative then?

Reply:
Set the token `expires_at` to something short, or change this line in oauth.js to run the check every 30s (for example).
`var TOKEN_EXPIRATION_ALLOWANCE = 119.5 * 60 * 1000;`

Reply:
I'm confused. If you mean "hack the local copy of the client in node_modules", I'm already doing that; if you don't mean that, I don't know what you mean, sorry.

Reply:
That's exactly right. Hack the installed copy of oauth.js so that the client checks with Panoptes for a session, rather than using the stored token.

Don't forget to block third-party cookies too, since that's the bug you're addressing in this PR.